src/current/advisories/a99561.md

```diff
@@ -0,0 +1,51 @@
+
+In CockroachDB v22.2 and later, the implementation for [`ALTER TABLE`](../v22.2/alter-table.html#add-column) began using the new [declarative schema changer](../v23.1/online-schema-changes.html#declarative-schema-changer) framework. When adding a new column using the new declarative schema changer, CockroachDB retrieves a unique secondary index mistakenly, assuming it to be a primary index. As a result, the secondary index is modified to store the new columns. CockroachDB expects there to be only one primary index, but in the case outlined here, there are two, because a secondary index has been incorrectly labeled as a primary one. This will cause the new column data to be present only in the secondary index and could lead to irrecoverable data loss, if the secondary index is deleted by the user.
+
+## Statement
+
+This is resolved in CockroachDB by [#105828](https://github.com/cockroachdb/cockroach/pull/105828), which makes CockroachDB automatically repair the encoding type for secondary indexes when accessing the table. This PR will also add validation to detect tables with this same problem so that they can be fixed.
+
+The fix will be applied to v22.2.12 and v23.1.6.
+
+This public issue is tracked by [#99561](https://github.com/cockroachdb/cockroach/issues/99561).
+
+## Mitigation
+
+Users of CockroachDB v22.2.x are encouraged to upgrade to [v22.2.12](../releases/v22.2.html) upon its release. Similarly, users of v23.1.0 — v23.1.5 are encouraged to update to [v23.1.6](../releases/v23.1.html) upon its release.
+
+Until such an upgrade, users can temporarily disable the declarative schema changer before executing an [`ALTER TABLE..ADD COLUMN`](../v22.2/alter-table.html#add-column) statement using the [session variable](../v23.1/set-vars.html):
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+SET use_declarative_schema_changer='off'
+~~~
+
+To disable the declarative schema changer for all users, use the [cluster setting](../v23.1/cluster-settings.html):
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+SET CLUSTER SETTINGS sql.defaults.use_declarative_schema_changer = off
+~~~
+
+Additionally, run the corruption detection query outlined in [this gist](https://gist.github.com/rimadeodhar/752bdf89d69c9d7fea3be5ed9acce3b6) to determine if there is any such secondary index on the cluster with primary key encoding. If such corruption exists, disable the declarative schema changer as outlined previously prior to running any schema change and contact our [support team](https://support.cockroachlabs.com/) to fix the cluster as soon as possible.
+
```

Comment:
```suggestion
Until such an upgrade, please run the query outlined in [this gist](https://gist.github.com/fqazi/2b71b05ce8f6b028267ed3b4498c4f6a) to check if your cluster is in a potentially corrupt state. If the query returns any data, please reach out to support for help to resolve the inconsistency. Until support has had a chance to fix your cluster, please refrain from running any schema changes on the impacted table including DROP INDEX or ALTER TABLE as it could lead to data loss.

```